src/Microsoft.AspNetCore.Razor.Sdk/build/netstandard2.0/Razor.Sdk.CurrentVersion.targets

```diff
@@ -185,10 +185,29 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
 
     <ItemGroup>
-      <RazorGenerate Update="@(RazorGenerate)" Condtion="'%(RazorGenerate.GeneratedOutput)'==''">
+      <!-- Calculating the path of the generated output file in the following order
+      1. Use GeneratedOutput specified on the Item if available
+      2. Use the value of Link if specified.
+      3. Calculate it using RelativeDir
+      -->
+      <RazorGenerate Condition="'%(RazorGenerate.GeneratedOutput)'=='' AND '%(RazorGenerate.Link)' != ''">
+        <GeneratedOutput>$(RazorGenerateIntermediateOutputPath)$([System.IO.Path]::ChangeExtension('%(RazorGenerate.Link)', '.cs'))</GeneratedOutput>
+      </RazorGenerate>
+
+      <RazorGenerate Condition="'%(RazorGenerate.GeneratedOutput)'==''">
         <GeneratedOutput>$(RazorGenerateIntermediateOutputPath)%(RelativeDir)%(Filename).cs</GeneratedOutput>
       </RazorGenerate>
     </ItemGroup>
+
+    <ItemGroup>
+      <RazorGenerate Condition="'%(RazorGenerate.ViewEnginePath)' == '' AND '%(RazorGenerate.Link)' != ''">
+        <ViewEnginePath>%(RazorGenerate.Link)</ViewEnginePath>
```

Comment:
I want to come up with another name for this. ViewEngine is an MVC-ism. Ideas?

Reply:
`RelativePath` \ `PhysicalRelativePath` \ `RootRelativePath`?

Reply:
`RelativePath` or `RelativePhysicalPath`. I like `RelativePath` a bit better